core/include/thread_flags.h

```diff
@@ -61,13 +65,40 @@
  * @name reserved thread flags
  * @{
  */
+
+#if defined(MODULE_CORE_THREAD_FLAGS) || defined(DOXYGEN)
+/**
+ * @brief Set when a message becomes available
+ *
+ * This flag is set for a thread if a message becomes available either by being
+ * queued in the thread's message queue or by another thread becoming
+ * send-blocked.
+ *
+ * It is only reset through thread_flags_wait_*(), not by msg_receive().
+ * Care must be taken to a) not lose message notifications and b) not block for
+ * messages but not thread flags.
+ * Use e.g., the following pattern when waiting for both thread flags and messages:
+ *
+ *      while(1) {
+ *          thread_flags_t flags = thread_flags_wait_any(0xFFFF);
+ *          if (flags & THREAD_FLAG_MSG_WAITING) {
+ *              msg_t msg;
+ *              while (msg_try_receive(&msg) == 1) {
```

Comment:
Other than the mask and using > 0 instead of == 1, I used this exact same loop for my event loop after writing #7533

Reply:
Hopefully we'll see this pattern in more places throughout the code base. ;)